File: server/app/Domain/DTOs/UserDTO.py
from dataclasses import dataclass
from datetime import date
from typing import Optional
from app.Domain.enums.role import Role

@dataclass
class UserDTO:
    id: int
    first_name: str
    last_name: str
    email: str
    date_of_birth: date
    gender: str
    country: str
    street: str
    street_number: str
    role: Role
    profile_image: Optional[str]
    created_at: str
    updated_at: Optional[str]

    @staticmethod
    def from_model(user) -> "UserDTO":
        return UserDTO(
            id=user.id,
            first_name=user.first_name,
            last_name=user.last_name,
            email=user.email,
            date_of_birth=user.date_of_birth,
            gender=user.gender,
            country=user.country,
            street=user.street,
            street_number=user.street_number,
            role=user.role,
            profile_image=user.profile_image,
            created_at=user.created_at.strftime("%Y-%m-%d %H:%M:%S"),
            updated_at=user.updated_at.strftime("%Y-%m-%d %H:%M:%S") if user.updated_at else None
        )

File: client/src/models/user/UserDTO.ts
export interface UserDTO {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  dateOfBirth: string | null;
  gender: string | null;
  country: string | null;
  street: string | null;
  streetNumber: string | null;
  role: string;
  profileImage: string | null;
  createdAt: string;
  updatedAt: string | null;
}

// Backend koristi snake_case
export interface UserDTOBackend {
  id: number;
  first_name: string;
  last_name: string;
  email: string;
  date_of_birth: string | null;
  gender: string | null;
  country: string | null;
  street: string | null;
  street_number: string | null;
  role: string;
  profile_image: string | null;
  created_at: string;
  updated_at: string | null;
}

export function mapUserDTOFromBackend(backend: UserDTOBackend): UserDTO {
  return {
    id: backend.id,
    firstName: backend.first_name,
    lastName: backend.last_name,
    email: backend.email,
    dateOfBirth: backend.date_of_birth,
    gender: backend.gender,
    country: backend.country,
    street: backend.street,
    streetNumber: backend.street_number,
    role: backend.role,
    profileImage: backend.profile_image,
    createdAt: backend.created_at,
    updatedAt: backend.updated_at,
  };
}